workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,460 @@
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the website-cert-friendly-name and remove the environments that are not used.
+      - name: website-cert-friendly-name
+        id: website-cert-friendly-name
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'WEBSITE_CERT_FRIENDLY_NAME'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the website-cert-password and remove the environments that are not used.
+      - name: website-cert-password
+        id: website-cert-password
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'WEBSITE_CERT_PASSWORD'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-service-user for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-service-user
+        id: iis-server-service-user
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_SERVICE_USER'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-service-password for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-service-password
+        id: iis-server-service-password
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_SERVICE_PASSWORD'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
+
+      # TODO:  Fill in the iis-server-cert-path for WinRm access to the IIS Server and remove the environments that are not used.
+      - name: iis-server-cert-path
+        id: iis-server-cert-path
+        uses: im-open/set-variable-based-on-environment@v1.0.0
+        with:
+          variable-name: 'IIS_SERVER_CERT_PATH'
+          current-environment: ${{ needs.set-vars.outputs.ENVIRONMENT }}
+          dev-value: ''
+          qa-value: ''
+          stage-value: ''
+          prod-value: ''
+          demo-value: ''
+          uat-value: ''
```

Comment:
I think some of these variables like this one and others (website-cert-friendly-name, website-cert-path, app-pool-name) have the potential to be the same for each environment.  In those cases I would actually make a global env variable and put the TODO there to fill them out or put it in the set-vars job if it varies between environment.  If you don't want to make that change, I'd at another TODO at the top of each applicable step to convert it to a global ENV if it is the same in each env.

Reply:
I'll add a TODO for the devs to figure out which vars to make global.  Things like teh cert friendly name may change from env to evn, so it's best for them to decide what stays the same.